src/platform/linux/mod.rs

```diff
@@ -386,76 +390,84 @@ impl OsIpcChannel {
 }
 
 pub struct OsIpcReceiverSet {
-    pollfds: Vec<pollfd>,
-}
-
-impl Drop for OsIpcReceiverSet {
-    fn drop(&mut self) {
-        unsafe {
-            for pollfd in self.pollfds.iter() {
-                let result = libc::close(pollfd.fd);
-                assert!(thread::panicking() || result == 0);
-            }
-        }
-    }
+    poll: Poll,
+    pollfds: HashMap<Token, c_int, BuildHasherDefault<FnvHasher>>,
+    events: Events
 }
 
 impl OsIpcReceiverSet {
     pub fn new() -> Result<OsIpcReceiverSet,UnixError> {
+        let fnv = BuildHasherDefault::<FnvHasher>::default();
         Ok(OsIpcReceiverSet {
-            pollfds: Vec::new(),
+            poll: Poll::new().expect("Could not create a new receiver"),
+            pollfds: HashMap::with_hasher(fnv),
+            events: Events::with_capacity(1024)
```

Comment:
Yeah, it's also in the main example. Still, since this is a pretty big number that feels like it might actually cause some noticeable overhead, I think it's worth checking whether it *needs* to be that big... (And whether it affects benchmark results.)